functions/responsestreaming/StreamBigQuery/Function.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_response_streaming]
+
+using Google.Cloud.Functions.Framework;
+using Microsoft.AspNetCore.Http;
+using System.Threading.Tasks;
+using Google.Cloud.BigQuery.V2;
+using System;
+
+namespace StreamBigQuery;
+
+public class Function : IHttpFunction
+{
+    public async Task HandleAsync(HttpContext context)
+    {
+        string projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        BigQueryClient client = BigQueryClient.Create(projectId).WithDefaultLocation(Locations.UnitedStates);
+        // Example to retrieve a large payload from BigQuery public dataset
+        string query = "SELECT abstract FROM `bigquery-public-data.breathe.bioasq` LIMIT 1000";
+        
+        BigQueryJob job = client.CreateQueryJob(
```

Comment:
Is there any reason for explicitly creating the query job and then fetching the results instead of using `client.ExecuteSql`?

Taking a step back - is there any reason we really need to use BigQuery at all for this? It would be really simple to have:

```csharp
for (int i = 0; i < 10; i++)
{
    await context.Response.WriteAsync($"Line {i}\n");
    await Task.Delay(1000);
}
```

That would demonstrate response streaming, without customers having to know anything about BigQuery.

If the decision has been made that response streaming samples should all use BigQuery, then okay - but if that decision is still up for review, I'd personally try to simplify.

Reply:
Yes, it was decided to use BigQuery for all the streaming samples